bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/core/AbstractVertex.java

```diff
@@ -126,40 +126,86 @@ public boolean isSource() {
+
     /**
      * Returns a list of data flow variables with the given name
      * <p>
-     * See {@link AbstractVertex#getNodeCharacteristicsWithName(String)} for a similar method for node characteristics
+     * See {@link AbstractVertex#getNodeCharacteristics(String)} for a similar method for node characteristics
      * @param dataFlowVariable Name of the data flow variable
      * @return Returns a list of all data flow variables with the given name
      */
-    public List<DataFlowVariable> getDataFlowVariablesWithName(String dataFlowVariable) {
+    public List<DataFlowVariable> getDataFlowVariables(String dataFlowVariable) {
         return this.getAllIncomingDataFlowVariables().stream()
                 .filter(it -> it.getVariableName().equals(dataFlowVariable))
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Returns a list of data flow variable names with the given name
+     * <p>
+     * See {@link AbstractVertex#getNodeCharacteristicNames(String)} for a similar method for node characteristics
+     * @param dataFlowVariable Name of the data flow variable
+     * @return Returns a list of all data flow variables with the given name
+     */
+    public List<String> getDataFlowVariableNames(String dataFlowVariable) {
+        return this.getAllIncomingDataFlowVariables().stream()
+                .filter(it -> it.getVariableName().equals(dataFlowVariable))
+                .map(DataFlowVariable::variableName)
+                .collect(Collectors.toList());
+    }
+
     /**
      * Returns a map containing the data flow characteristics with the given characteristic type for each data flow variable
+     * <p>
+     * To get the characteristic value names for the data flow variables use {@link AbstractVertex#getDataFlowCharacteristicNames(String)}
      * @param characteristicType Name of the characteristic type
      * @return Returns a map with data flow characteristics with the given name for each data flow variable
      */
-    public Map<String, List<CharacteristicValue>> getDataFlowCharacteristicsWithName(String characteristicType) {
+    public Map<String, List<CharacteristicValue>> getDataFlowCharacteristics(String characteristicType) {
         return this.getAllIncomingDataFlowVariables().stream()
                 .collect(Collectors.toMap(DataFlowVariable::getVariableName, it -> it.getCharacteristicsWithName(characteristicType)));
     }
 
+    /**
+     * Returns a map containing the data flow characteristics with the given characteristic type for each data flow variable
+     * <p>
+     * To get the characteristic values for the data flow variables use {@link AbstractVertex#getDataFlowCharacteristics(String)}
+     * @param characteristicType Name of the characteristic type
+     * @return Returns a map with data flow characteristics with the given name for each data flow variable
+     */
+    public Map<String, List<String>> getDataFlowCharacteristicNames(String characteristicType) {
```

Comment:
I think we called them data characteristics elsewhere (instead of data flow characteristics) as they represent information about the data as is, and not the flow.

Reply:
Fixed in 0cfeb97e7b29a1fcacb1159584166cc2a462e692